pkg/logging/logging.go

```diff
@@ -1,140 +1,115 @@
-// stacktraces are added more liberally
-func WithDevelopmentMode(b bool) Option {
-	return func(lc *loggingConfig) {
-		lc.Config.Development = b
-	}
-}
-
-// WithStackTraces enables/disables stacktraces logged under the "stacktraces" key
-func WithStackTraces(b bool) Option {
+// WithOutput sets the log output
+func WithOutput(w io.Writer) Option {
 	return func(lc *loggingConfig) {
-		lc.Config.DisableStacktrace = !b
+		lc.output = w
 	}
 }
 
-// WithOutputPaths sets to which paths level "info" and above shall be sent. Treats "stdout" and "stderr" as
-// special paths
-func WithOutputPaths(paths []string) Option {
+// WithCaller sets whether the calling source should be logged, since the operation is
+// computationally expensive
+func WithCaller(b bool) Option {
 	return func(lc *loggingConfig) {
-		lc.Config.OutputPaths = paths
-	}
-}
-
-// WithErrorPaths sets to which paths level "error" and below shall be sent. Treats "stdout" and "stderr" as
-// special paths
-func WithErrorPaths(paths []string) Option {
-	return func(lc *loggingConfig) {
-		lc.Config.ErrorOutputPaths = paths
+		lc.enableCaller = b
 	}
 }
 
 // Init initializes the global logger. The `encoding` variable sets whether content should
 // be printed for console output or in JSON (for machine consumption)
-func Init(appName, appVersion, logLevel, encoding string, opts ...Option) error {
-	switch strings.ToLower(encoding) {
-	case "json", "console", "logfmt":
-	default:
-		return fmt.Errorf("unknown encoding %q", encoding)
-	}
-
-	var level zap.AtomicLevel
-	switch strings.ToLower(logLevel) {
-	case "debug":
-		level = zap.NewAtomicLevelAt(zapcore.DebugLevel)
-	case "info":
-		level = zap.NewAtomicLevelAt(zapcore.InfoLevel)
-	case "warn":
-		level = zap.NewAtomicLevelAt(zapcore.WarnLevel)
-	case "error":
-		level = zap.NewAtomicLevelAt(zapcore.ErrorLevel)
-	case "fatal":
-		level = zap.NewAtomicLevelAt(zapcore.FatalLevel)
-	case "panic":
-		level = zap.NewAtomicLevelAt(zapcore.PanicLevel)
-	default:
-		return fmt.Errorf("unsupported log level %q", logLevel)
-	}
-
-	zapEncoderConfig := zapcore.EncoderConfig{
-		TimeKey:        "ts",
-		LevelKey:       "level",
-		NameKey:        "logger",
-		CallerKey:      "caller",
-		MessageKey:     "msg",
-		StacktraceKey:  "stacktrace",
-		LineEnding:     zapcore.DefaultLineEnding,
-		EncodeLevel:    zapcore.LowercaseLevelEncoder,
-		EncodeTime:     zapcore.ISO8601TimeEncoder,
-		EncodeDuration: zapcore.MillisDurationEncoder,
-		EncodeCaller:   zapcore.ShortCallerEncoder,
-	}
-
-	// registering the encoder is taken care of the by teh zaplogfmt library itself
-	if encoding == "logfmt" {
-		zapEncoderConfig.EncodeTime = func(ts time.Time, encoder zapcore.PrimitiveArrayEncoder) {
-			encoder.AppendString(ts.UTC().Format(time.RFC3339))
+func Init(version, logLevel, encoding string, opts ...Option) error {
+	var level = new(slog.Level)
+	err := level.UnmarshalText([]byte(logLevel))
```

Comment:
I don't know what it is, but this pattern `slog` uses confuses me: I get that there should be a convenience function that parses the stuff users put in as good as possible, but shouldn't the API itself stick to an enumerated set of levels? If the calling code (e.g. `goConvert` or whatever) wants to allow users to try their luck that's totally cool of course, but IMHO it's weird to do that inside the initializer. Am I being stupid here?

Reply:
Not stupid. As said above: it's more convenient for the caller than having to figure it out all the time in every tool (copy-pasta galore)